lib/rubocop/target_ruby.rb

```diff
@@ -44,33 +44,66 @@ def find_version
     # The target ruby version may be found in a .ruby-version file.
     # @api private
     class RubyVersionFile < Source
-      FILENAME = '.ruby-version'
+      RUBY_VERSION_FILENAME = '.ruby-version'
+      # rubocop:disable Lint/MixedRegexpCaptureTypes
+      # `(ruby-)` is not a capture type.
+      RUBY_VERSION_PATTERN = /\A(ruby-)?(?<version>\d+\.\d+)/.freeze
+      # rubocop:enable Lint/MixedRegexpCaptureTypes
```

Comment:
I believe this should avoid the need to disable the cop:

```suggestion
      RUBY_VERSION_PATTERN = /\A(?:ruby-)?(?<version>\d+\.\d+)/.freeze
```

Reply:
Thanks! Pushed a `fixup!` commit with your recommendation (here and in `RuboCop::TargetRuby::ToolVersionsFile`)

Reply:
(was https://github.com/rubocop-hq/rubocop/pull/9319/commits/1ee6035b30fc8acbfd815a1ecbe1b7d6fdb7b086, squashed and rebased)